include/physfs.hpp

```diff
@@ -8,171 +8,116 @@
+void setAllocator(Allocator const * allocator);
+void mount(std::string const & newDir, std::string const & mountPoint, bool appendToPath);
 
-void setSaneConfig(string const & orgName, string const & appName, string const & archiveExt, bool includeCdRoms, bool archivesFirst);
-
-void mkdir(string const & dirName);
-
-void deleteFile(string const & filename);
-
-string getRealDir(string const & filename);
-
-StringList enumerateFiles(string const & directory);
-
-void enumerateFiles(string const & directory, EnumFilesCallback callback, void * extra);
-
-bool exists(string const & filename);
-
-bool isDirectory(string const & filename);
-
-bool isSymbolicLink(string const & filename);
+std::string getDirSeparator();
+std::string getBaseDir();
+std::string getUserDir();
+std::string getWriteDir();
+std::string getRealDir(std::string const & filename);
+std::string getMountPoint(std::string const & dir);
 
-sint64 getLastModTime(string const & filename);
+StringList getCdRomDirs();
+StringList getSearchPath();
+StringList enumerateFiles(std::string const & directory);
 
+bool exists(std::string const & filename);
+bool isDirectory(std::string const & filename);
+bool isSymbolicLink(std::string const & filename);
 bool isInit();
-
 bool symbolicLinksPermitted();
 
-void setAllocator(Allocator const * allocator);
-
-void mount(string const & newDir, string const & mountPoint, bool appendToPath);
+ArchiveInfoList supportedArchiveTypes();
 
-string getMountPoint(string const & dir);
+sint64 getLastModTime(std::string const & filename);
 
 namespace Util {
-
-sint16 swapSLE16(sint16 value);
-
-uint16 swapULE16(uint16 value);
-
-sint32 swapSLE32(sint32 value);
-
-uint32 swapULE32(uint32 value);
-
-sint64 swapSLE64(sint64 value);
-
-uint64 swapULE64(uint64 value);
-
-sint16 swapSBE16(sint16 value);
-
-uint16 swapUBE16(uint16 value);
-
-sint32 swapSBE32(sint32 value);
-
-uint32 swapUBE32(uint32 value);
-
-sint64 swapSBE64(sint64 value);
-
-uint64 swapUBE64(uint64 value);
-
-string utf8FromUcs4(uint32 const * src);
-
-string utf8ToUcs4(char const * src);
-
-string utf8FromUcs2(uint16 const * src);
-
-string utf8ToUcs2(char const * src);
-
-string utf8FromLatin1(char const * src);
-
+    sint16 swapSLE16(sint16 value);
+    uint16 swapULE16(uint16 value);
+    sint32 swapSLE32(sint32 value);
+    uint32 swapULE32(uint32 value);
+    sint64 swapSLE64(sint64 value);
+    uint64 swapULE64(uint64 value);
+    sint16 swapSBE16(sint16 value);
+    uint16 swapUBE16(uint16 value);
+    sint32 swapSBE32(sint32 value);
+    uint32 swapUBE32(uint32 value);
+    sint64 swapSBE64(sint64 value);
+    uint64 swapUBE64(uint64 value);
+
+    std::string utf8FromUcs4(uint32 const * src);
+    std::string utf8ToUcs4(char const * src);
+    std::string utf8FromUcs2(uint16 const * src);
+    std::string utf8ToUcs2(char const * src);
+    std::string utf8FromLatin1(char const * src);
 }
 
 }
 
-#endif /* _INCLUDE_PHYSFS_HPP_ */
+#endif /* _INCLUDE_PHYSFS_HPP_ */
```

Comment:
I like these changes, and will include probably include them verbatim, once other issues are addressed (or if you did a pull request with just these).